tck/src/main/java/org/eclipse/microprofile/fault/tolerance/tck/AsynchronousTest.java

```diff
@@ -109,6 +112,7 @@ public void testClassLevelAsyncIsFinished() {
         Future<Connection> future = clientClass.service(waitingFuture);
         await().atMost(2000, TimeUnit.MILLISECONDS)
             .untilAsserted(()-> Assert.assertTrue(future.isDone()));
+        waitingFuture.complete(null);
```

Comment:
Indeed the way you added the `future.complete()` calls is equivalent to the `@AfterTest` method and is completely unnecessary.

What I'm saying is that the `test*AsyncIsFinished` tests should complete the waiting future _before_ the assert. If you don't do that, the test will needlessly wait one second.

In fact, I also think that your should revert the timeout in the `serviceCS` method back to 1000 seconds, to make sure all callers actually complete the waiting future instead of relying on the timeout.

Reply:
Except for `testAsyncIsNotFinished` which doesn't need to complete the future (or indeed wait for anything)

Reply:
Thanks @Ladicek I will make the changes accordingly

Reply:
@Azquelt is of course right -- I specifically said `test*AsyncIsFinished`, which doesn't include the `*NotFinished` tests. Those should not complete the future, and indeed don't have to wait for anything.

Reply:
Doh, I should read more carefully :wink:

Reply:
Done guys, thanks for your feedback